workers/unity/Packages/io.improbable.gdk.transformsynchronization/TransformUtils.cs

```diff
@@ -66,22 +66,35 @@ public static CompressedQuaternion ToCompressedQuaternion(this Quaternion quater
         ///     Constructs an <see cref="EntityQueryDesc"/> given an array of base types and
         ///     a generic component type.
         /// </summary>
+        /// <param name="requireAuthority">If Transform Authority is required for this query</param>
         /// <param name="baseTypes">The base set of types.</param>
         /// <typeparam name="T">The type to add.</typeparam>
         /// <returns>An <see cref="EntityQueryDesc"/> that is the union of <see cref="baseTypes"/> and typeof(<see cref="T"/>)</returns>
-        internal static EntityQueryDesc ConstructEntityQueryDesc<T>(params ComponentType[] baseTypes)
+        internal static EntityQueryDesc ConstructEntityQueryDesc<T>(bool requireAuthority, params ComponentType[] baseTypes)
         {
             var componentType = ComponentType.ReadOnly<T>();
-
             var includedComponentTypes = baseTypes
-                .Append(componentType)
-                .ToArray();
+                .Append(componentType);
+
+            if (requireAuthority)
+            {
+                includedComponentTypes = includedComponentTypes
+                    .Append(ComponentType.ReadOnly<TransformInternal.Authoritative>());
+            }
 
             var componentQueryDesc = new EntityQueryDesc
             {
-                All = includedComponentTypes
+                All = includedComponentTypes.ToArray(),
             };
 
+            if (!requireAuthority)
+            {
+                componentQueryDesc.None = new[]
+                {
+                    ComponentType.ReadOnly<TransformInternal.Authoritative>()
+                };
+            }
+
             return componentQueryDesc;
```

Comment:
How useful is this function now? Like effectively all it does is add `T` to all and then the `TransformInternal.Authoritative` to either the `All` or `None` depending on the bool (which is 😕 already)

Reply:
I'm considering replacing this whole thing, but I think thats for later.